packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -80,6 +81,18 @@ class TextPainter {
     _needsLayout = true;
   }
 
+  /// The string used to ellipsize overflowing text.
```

Comment:
Clarified the docs and added an assert for empty ellipsis strings

(I don't know the internals of the text rendering engine well enough to precisely describe its behavior)